exporters/prometheus/exporter.go

```diff
@@ -124,16 +127,31 @@ func (c *collector) Collect(ch chan<- prometheus.Metric) {
 		}
 	}
 
-	c.createTargetInfoOnce.Do(func() {
-		// Resource should be immutable, we don't need to compute again
-		targetInfo, err := c.createInfoMetric(targetInfoMetricName, targetInfoDescription, metrics.Resource)
-		if err != nil {
-			// If the target info metric is invalid, disable sending it.
-			otel.Handle(err)
-			c.disableTargetInfo = true
+	// Initialize (once) targetInfo and disableTargetInfo.
+	func() {
+		var err error
+		var targetInfo prometheus.Metric
+
+		c.mu.Lock()
+		defer func() {
+			c.mu.Unlock()
+			if err != nil {
+				otel.Handle(err)
+			}
+		}()
+
+		if c.targetInfo == nil && !c.disableTargetInfo {
+			targetInfo, err = createInfoMetric(targetInfoMetricName, targetInfoDescription, metrics.Resource)
+			if err != nil {
+				// If the target info metric is invalid, disable sending it.
+				c.disableTargetInfo = true
+				return
+			}
+
+			c.targetInfo = targetInfo
```

Comment:
```suggestion
	func() {
		c.mu.Lock()
		defer c.mu.Unlock()

		if c.targetInfo == nil && !c.disableTargetInfo {
			targetInfo, err := createInfoMetric(targetInfoMetricName, targetInfoDescription, metrics.Resource)
			if err != nil {
				// If the target info metric is invalid, disable sending it.
				c.disableTargetInfo = true
				otel.Handle(err)
				return
			}

			c.targetInfo = targetInfo
```
Simplify this block.